lib/modules/archivist/ArchivistModuleError.js

```diff
@@ -0,0 +1,19 @@
+// This module is a gateway between frontend's and the archivist in the
+// MAGE backend.
+const MageError = require('lib/mage').MageError;
+
+class ArchivistModuleError extends MageError {
+	constructor(message, state, details) {
+		super({
+			message: message,
+			details: Object.assign({
+				actorId: state.actorId,
+				acl: state.acl
+			}, details)
+		});
+
+		this.type = ArchivistModuleError;
```

Comment:
What's the purpose of this? `error.constructor` === `error.type`

Reply:
Ah, I think that should be a string. Ill fix it

Reply:
Then I would suggest having `MageError` automatically assign `this.type = this.constructor.name;`, so you never have to type that line of code again.